rust/src/pgsql/parser.rs

```diff
@@ -1049,6 +1060,22 @@ pub fn parse_copy_out_response(i: &[u8]) -> IResult<&[u8], PgsqlBEMessage, Pgsql
     ))
 }
 
+pub fn parse_copy_in_response(i: &[u8]) -> IResult<&[u8], PgsqlBEMessage, PgsqlParseError<&[u8]>> {
+    let (i, identifier) = verify(be_u8, |&x| x == b'G')(i)?;
```

Comment:
Still feels like double parsing

Reply:
I'm considering this as part of the future debt from the other comment, too. I'll take time to better grasp Rust Generics, and once we're past Release schedule, will come back to this.